### PR TITLE
Move MRPC_PART_INFO out of for loop to improve code efficiency

### PR DESCRIPTION
**Pull request by xyan264**
Move MRPC_PART_INFO command up one level: from switchtec_fw_part_info_gen4() to switchtec_fw_part_info(), out of the for loop.

### CODE REVIEW

File: lib/fw.c [hunk 1]
```diff
@@ -970,9 +962,20 @@ static int switchtec_fw_part_info(struct switchtec_dev *dev, int nr_info,
...
+	struct switchtec_flash_info_gen4 all_info;
+
 	if (info == NULL || nr_info == 0)
 		return -EINVAL;
 
+	if (dev->gen == SWITCHTEC_GEN4) {
+		ret = switchtec_cmd(dev, MRPC_PART_INFO, &subcmd,
+				    sizeof(subcmd), &all_info,
+				    sizeof(all_info));
+		if (ret < 0)
```
Comment (kelvin-cao) [inline lib/fw.c]: Any non-zero value should be taken as error.

File: lib/fw.c [hunk 2]
```diff
@@ -970,9 +962,20 @@ static int switchtec_fw_part_info(struct switchtec_dev *dev, int nr_info,
 	int ret;
 	int i;
 
+	uint8_t subcmd = MRPC_PART_INFO_GET_ALL_INFO;
```
Comment (kelvin-cao) [inline lib/fw.c]: Unnecessary blank line.

File: lib/fw.c (hunk 1 above)
Comment (xyan264) [reply lib/fw.c]: Updated, please review.

File: lib/fw.c (hunk 2 above)
Comment (xyan264) [reply lib/fw.c]: Updated, please review.

**Pull request merged**